Crypto/PublicKey+Apple.mm

```diff
@@ -682,16 +686,36 @@ virtual int _sign(int/*mbedtls_md_type_t*/ mbedDigestAlgorithm,
             
             CFIndex count = SecTrustGetCertificateCount(trustRef);
             Assert(count > 0);
+            if (count == 1) {
+                NSDictionary* params = @{
+                    (id)kSecClass:              (id)kSecClassCertificate,
+                    (id)kSecValueRef:           (__bridge id)certRef
+                };
+                checkOSStatus(SecItemDelete((CFDictionaryRef)params),
+                              "SecItemDelete",
+                              "Couldn't delete a certificate from the Keychain");
+                return;
+            }
 #if __MAC_OS_X_VERSION_MAX_ALLOWED >= 120000 || __IPHONE_OS_VERSION_MAX_REQUIRED >= 150000
             if (@available(macOS 12.0, iOS 15.0, *)) {
                 CFArrayRef certs = SecTrustCopyCertificateChain(trustRef);
                 for (CFIndex i = count - 1; i >= 0; i--) {
-                    SecCertificateRef ref = (SecCertificateRef)CFArrayGetValueAtIndex(certs, i);
-                    if (getChildCertCount(ref) < 2) {
+                    SecCertificateRef copiedRef = (SecCertificateRef)CFArrayGetValueAtIndex(certs, i);
+                    if (getChildCertCount(copiedRef) < 2) {
+                        // CertRef returned from CopyCertificateChain won't get deleted, so we fetch directly. 
+                        NSData* certData = CFBridgingRelease(findInKeychain(@{
+                            (id)kSecClass:              (id)kSecClassCertificate,
+                            (id)kSecValueRef:           (__bridge id)copiedRef,
+                            (id)kSecAttrLabel:          label,
```

Comment:
This might not be correct as the chain (parent) cert will not have the label. I think only value ref is enough?